azurerm/resource_arm_postgresql_firewall_rule_test.go

```diff
@@ -0,0 +1,124 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMPostgreSQLFirewallRule_basic(t *testing.T) {
+	resourceName := "azurerm_postgresql_firewall_rule.test"
+	ri := acctest.RandInt()
+	config := testAccAzureRMPostgreSQLFirewallRule_basic(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPostgreSQLFirewallRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPostgreSQLFirewallRuleExists(resourceName),
```

Comment:
Can we add some state checks here, please?

Reply:
as discussed on Slack - we tend to lean on the Import tests for this - so I'm not sure this is needed, but I've added some anyway